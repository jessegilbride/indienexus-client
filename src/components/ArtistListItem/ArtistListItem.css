a.ArtistListItem {
  display: block;
  max-width: 800px;
  margin: 1em auto;
  padding-left: 1em;
  padding-bottom: 1em;
  border: 1px solid lightgrey;
  text-decoration: none;
}

a.ArtistListItem:hover {
  outline: 1px solid darkslategrey;
}
