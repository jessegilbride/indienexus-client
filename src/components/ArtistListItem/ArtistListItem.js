import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { NiceDate } from '../Utils/Utils';
import './ArtistListItem.css';

export default class ArtistListItem extends Component {
  render() {
    /* id, name, bio, soundcloud_embed, tags, date_created */
    const { artist } = this.props;
    // console.log(artist);
    return (
      <Link to={`/artist/${artist.id}`} className='ArtistListItem'>
        <header className='ArtistListItem__header'>
          <h3 className='ArtistListItem__heading'>{artist.name}</h3>
        </header>
        <div className='ArtistListItem__footer'>
          {/* <div><ArtistTags artist={artist} /></div> */}
          <div><ArtistDate artist={artist} /></div>
        </div>
      </Link>
    );
  }
}

ArtistListItem.defaultProps = {
  artist: {}
}

/* function ArtistTags({ artist }) {
  return (
    <span className=''>
      Tags: {artist.tags}
    </span>
  );
} */

function ArtistDate({ artist }) {
  return (
    <span className='ArtistListItem__date'>
      Joined: <NiceDate date={artist.date_created} />
    </span>
  );
}

