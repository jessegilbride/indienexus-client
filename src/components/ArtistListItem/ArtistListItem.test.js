import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import ArtistListItem from './ArtistListItem';

// import ArtistListContext from '../../contexts/ArtistListContext';
// static contextType = ArtistListContext;

/* import ArtistContext from '../../contexts/ArtistContext';
// contextType = ArtistContext;
const context = ArtistContext;

const artist = {
  id: 1,
  user_id:1,
  name: 'First test artist',
  bio: 'Vestibulum id ligula porta felis euismod semper.',
  tags: 'tag1, tag2',
  soundcloud_embed: '<div>soundcloud</div>',
  date_created: new Date('2029-01-22T16:28:32.615Z')
}

// this.context.setArtist(artist);
context.setArtist(artist); */

describe('ArtistListItem component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <MemoryRouter><ArtistListItem /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <MemoryRouter><ArtistListItem name="ArtistListItem" /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});
