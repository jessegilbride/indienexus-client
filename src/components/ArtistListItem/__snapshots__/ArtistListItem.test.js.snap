// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistListItem component renders the UI as expected 1`] = `
<a
  className="ArtistListItem"
  href="/artist/undefined"
  onClick={[Function]}
>
  <header
    className="ArtistListItem__header"
  >
    <h3
      className="ArtistListItem__heading"
    />
  </header>
  <footer
    className="ArtistListItem__footer"
  >
    <div>
      <span
        className=""
      >
        Tags: 
      </span>
    </div>
    <div>
      <span
        className="ArtistListItem__date"
      >
        Joined: 
        Invalid Date
      </span>
    </div>
  </footer>
</a>
`;
